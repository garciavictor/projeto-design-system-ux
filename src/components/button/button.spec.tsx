import "@testing-library/jest-dom";
import { Button } from ".";
import { describe, it, expect, vi } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import { faGoogle } from "@fortawesome/free-brands-svg-icons";

describe("@components/Button", () => {
    it("should render correctly with default setup", () => {
        render(<Button buttonText="Click Me" onClick={() => { }} />);

        const button = screen.getByText("Click Me");
        expect(button).toBeInTheDocument();
    });

    it("should call onClick when clicked", () => {
        const onClick = vi.fn();
        render(<Button buttonText="Click Me" onClick={onClick} />);

        fireEvent.click(screen.getByRole("button"));
        expect(onClick).toHaveBeenCalledTimes(1);
    });

    it("should match snapshot", () => {
        const { asFragment } = render(<Button buttonText="Click Me" icon={faGoogle} onClick={() => { }} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
