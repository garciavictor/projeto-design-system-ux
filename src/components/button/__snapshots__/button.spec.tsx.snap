// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@components/button > should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bXdtCk hBrJEN"
  >
    <span
      class="sc-fmZSGO igSwgy"
    >
      Click Me
    </span>
  </button>
</DocumentFragment>
`;
